modules/barcode/samples/barcode.cpp

```diff
@@ -0,0 +1,302 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+// Copyright (c) 2020-2021 darkliang wangberlinT Certseeds
+
+#include <iostream>
+#include "opencv2/barcode.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/highgui.hpp"
+
+using namespace cv;
+using namespace std;
+
+static int liveBarCodeDetect();
+
+static int imageBarCodeDetect(const string &in_file);
+
+static bool g_detectOnly = false;
+
+static string g_out_file_name, g_out_file_ext;
+
+static Ptr<barcode::BarcodeDetector> bardet;
+
+int main(int argc, char **argv)
+{
+
+
+    const string keys = "{h help ? |        | print help messages }"
+                        "{i in     |        | input image path (also switches to image detection mode) }"
+                        "{detect   | false  | detect 1D barcode only (skip decoding) }"
+                        "{o out    |        | path to result file (only for single image decode) }"
+                        "{sr_prototxt|      | super resolution prototxt path }"
+                        "{sr_model |        | super resolution model path }";
+
+    CommandLineParser cmd_parser(argc, argv, keys);
+
+    cmd_parser.about("This program detects the 1D barcodes from camera or images using the OpenCV library.");
+    if (cmd_parser.has("help"))
+    {
+        cmd_parser.printMessage();
+        return 0;
+    }
+
+    string in_file_name = cmd_parser.get<string>("in");    // path to input image
+    string sr_prototxt = cmd_parser.get<string>("sr_prototxt");    // path to sr_prototxt
+    string sr_model = cmd_parser.get<string>("sr_model");    // path to sr_model
+    if (cmd_parser.has("out"))
+    {
+        std::string fpath = cmd_parser.get<string>("out");   // path to output image
+        std::string::size_type idx = fpath.rfind('.');
+        if (idx != std::string::npos)
+        {
+            g_out_file_name = fpath.substr(0, idx);
+            g_out_file_ext = fpath.substr(idx);
+        }
+        else
+        {
+            g_out_file_name = fpath;
+            g_out_file_ext = ".png";
+        }
+    }
+    if (!cmd_parser.check())
+    {
+        cmd_parser.printErrors();
+        return -1;
+    }
+    g_detectOnly = cmd_parser.has("detect") && cmd_parser.get<bool>("detect");
+    //! [initialize]
+    try{
+        bardet = makePtr<barcode::BarcodeDetector>(sr_prototxt, sr_model);
+    } catch (const std::exception& e)
+    {
+        cout <<
+             "\n---------------------------------------------------------------\n"
+             "Failed to initialize super resolution.\n"
+             "Please, download 'sr.*' from\n"
+             "https://github.com/WeChatCV/opencv_3rdparty/tree/wechat_qrcode\n"
```

Comment:
> wechat_qrcode

1. is this a correct message?
2. is this a mandatory prerequisite? (in QR codes it doesn't)

Reply:
1. I added that sentence:

> Or you can leave sr_prototxt and sr_model unspecified.

2. It is not a mandatory prerequisite.